ramda.d.ts

```diff
@@ -1630,12 +1630,15 @@ declare namespace R {
          * elements of each list.
          */
         union<T>(as: T[], bs: T[]): T[];
+        union<T>(as: T[]): (bs: T[]) => T[];
 
         /**
          * Combines two lists into a set (i.e. no duplicates) composed of the elements of each list.  Duplication is
          * determined according to the value returned by applying the supplied predicate to two list elements.
          */
         unionWith<T>(pred: (a: T, b: T) => boolean, list1: T[], list2: T[]): T[];
+        unionWith<T>(pred: (a: T, b: T) => boolean): (list1: T[], list2: T[]) => T[];
```

Comment:
This returning function need to be curried too. You could use CurriedFunction2 for this.